src/rust/engine/src/nodes.rs

```diff
@@ -1053,22 +1053,25 @@ impl Node for NodeKey {
 
   fn run(self, context: Context) -> NodeFuture<NodeResult> {
     let mut workunit_state = workunit_store::expect_workunit_state();
-    let maybe_started_workunit_id: Option<String> = if context.session.should_handle_workunits() {
-      self.user_facing_name().map(|node_name| {
-        let span_id = new_span_id();
-        let desc = self.display_info().and_then(|di| di.desc.as_ref().cloned());
 
-        // We're starting a new workunit: record our parent, and set the current parent to our span.
-        let parent_id = std::mem::replace(&mut workunit_state.parent_id, Some(span_id.clone()));
-        let metadata = WorkunitMetadata { desc };
+    let started_workunit_id = {
+      let display = context.session.should_handle_workunits() && self.user_facing_name().is_some();
+      let name = self.user_facing_name().unwrap_or(format!("{}", self));
+      let span_id = new_span_id();
+      let desc = self.display_info().and_then(|di| di.desc.as_ref().cloned());
+
+      // We're starting a new workunit: record our parent, and set the current parent to our span.
+      let parent_id = std::mem::replace(&mut workunit_state.parent_id, Some(span_id.clone()));
+      let metadata = WorkunitMetadata {
+        desc,
+        display,
+        blocked: false,
```

Comment:
I still think the "blocked" flag should be removed. There is really only "Started" and "Completed".

And I think that the presence of this flag is why I thought that you were mutating the metadata: it's not possible for this flag to stay accurate, because the workunit doesn't know when the work it wraps stops being "blocked" and starts running.

Reply:
In this schema, the "blocked" state is a static property of a given workunit. The engine records a span of time representing time spent blocked + time spent running, and then an inner span of time representing time spent running, and both of these are part of the overall record of how long a given pants job took and how much time was spent doing what. The flag just lets the V2 UI know that a given sink workunit (sink with respect to the workunit directed graph) represents this time blocked + time spent running state, and therefore its start time isn't meaningful enough to display in the v2 UI.